javascript/src/test/components/BasicCourseSearch/BasicCourseSearchForm.test.js

```diff
@@ -82,5 +82,55 @@ describe("BasicCourseSearchForm tests", () => {
 
     });
 
+    test("the download csv button seems to work", async () => {
+
+        const sampleReturnValue = {
+            "sampleKey": "sampleValue"
+        };
+
+        // Create spy functions (aka jest function, magic function)
+        // The function doesn't have any implementation unless
+        // we specify one.  But it does keep track of whether 
+        // it was called, how many times it was called,
+        // and what it was passed.
+
+        const setCourseJSONSpy = jest.fn();
+        const fetchJSONSpy = jest.fn();
+
+        fetchJSONSpy.mockResolvedValue(sampleReturnValue);
+
+        const { getByText, getByLabelText } = render(
+            <BasicCourseSearchForm setCourseJSON={setCourseJSONSpy} fetchJSON={fetchJSONSpy} />
+        );
+
+        const expectedFields = {
+            quarter: "20204",
+            department: "MATH",
+            level: "G"
+        };
+
+        const selectQuarter = getByLabelText("Quarter")
+        userEvent.selectOptions(selectQuarter, "20204");
+        const selectDepartment = getByLabelText("Department")
+        userEvent.selectOptions(selectDepartment, "MATH");
+        const selectLevel = getByLabelText("Course Level")
+        userEvent.selectOptions(selectLevel, "G");
+
+        const submitButton = getByText("Submit");
+        userEvent.click(submitButton);
+
+        // we need to be careful not to assert this expectation
+        // until all of the async promises are resolved
+        await waitFor(() => expect(setCourseJSONSpy).toHaveBeenCalledTimes(1));
+        await waitFor(() => expect(fetchJSONSpy).toHaveBeenCalledTimes(1));
+
+        // assert that ourSpy was called with the right value
+        expect(setCourseJSONSpy).toHaveBeenCalledWith(sampleReturnValue);
+        expect(fetchJSONSpy).toHaveBeenCalledWith(expect.any(Object), expectedFields);
+
+        // lets try the download csv now
+        // const csvButton = getByText("Download CSV");
+        // userEvent.click(csvButton);
```

Comment:
Is it necessary to keep this commented code in the final

Reply:
I agree with Andrew.. this should be deleted

Reply:
This code still seems to be here...

Reply:
But ok...